napari/_qt/qt_plugin_list.py

```diff
@@ -0,0 +1,199 @@
+"""Provides a QtPluginSorter that allows the user to change plugin call order.
+"""
+from qtpy.QtCore import Qt, Signal
+from qtpy.QtWidgets import (
+    QComboBox,
+    QDialog,
+    QListWidget,
+    QListWidgetItem,
+    QLabel,
+    QSizePolicy,
+    QVBoxLayout,
+    QWidget,
+)
+from ..plugins import plugin_manager as napari_plugin_manager
+from ..plugins.utils import permute_hook_implementations, HookOrderType
+from pluggy.manager import PluginManager, _HookCaller
+from typing import Optional
+
+
+class QtHookImplItem(QListWidgetItem):
+    """Item in a QtHookImplListWidget."""
+
+    def __init__(self, hookimpl, parent=None):
+        self.plugin_name = hookimpl.plugin_name
+        if self.plugin_name == 'builtins':
+            self.plugin_name = 'napari built-in'
+        super().__init__(self.plugin_name, parent)
+        self.hookimpl = hookimpl
+        self.setBackground(Qt.black)
+
+
+class QtHookImplListWidget(QListWidget):
+    """A QListWidget that allows sorting of plugin hooks."""
+
+    order_changed = Signal(list)  # emitted when the user changes the order.
+
+    def __init__(
+        self,
+        hook: Optional[_HookCaller] = None,
+        parent: Optional[QWidget] = None,
+    ):
+        """A ListWidget that shows all of the hook implementations for a spec.
+
+        Usually instantiated by QtPluginSorter.
+
+        Parameters
+        ----------
+        hook : pluggy.manager._HookCaller, optional
+            A pluggy HookCaller to show implementations for. by default None
+            (i.e. no hooks shown)
+        parent : QWidget, optional
+            Optional parent widget, by default None
+        """
```

Comment:
@tlambert03 refer to my comment on other PRs... =) ie this docstring should, tragically, be on the class docstring.